src/components/implementation/no_interface/llbooter/llbooter.c

```diff
@@ -108,24 +110,101 @@ boot_spd_end(struct cobj_header *h)
 	return sect->vaddr + round_up_to_page(sect->bytes);
 }
 
+struct cobj_symb *
+cobj_find_symb(const char *symbol_name, int cobj) {
+	unsigned int j = 0;
+	for (j = 0; j < hs[cobj]->nsymb; j++) {
+		struct cobj_symb *symb;
+
+		symb = cobj_symb_get(hs[cobj], j);
+		assert(symb);
+
+		if (!strcmp(symb->name, symbol_name) && symb->type == COBJ_SYMB_EXPORTED) {
+			// printc("found: %s, spdid: %d, addr: %x \n", symb->name, cobj, symb->vaddr);
+			return symb;
+		}
+				
+	}		
+	return NULL;
+}
+
+vaddr_t
+boot_find_invsymb_addr(struct cobj_symb *undefsymb) {
+	
+	unsigned int i;
+	vaddr_t symb_addr;	
+	struct cobj_symb *symb;
+	/* HACK is this the best way to do this? */
+	char jumper_symb[25];
+	strcpy(jumper_symb, undefsymb->name);
+	strcat(jumper_symb, "_inv");
+
```

Comment:
to avoid some of these problems, `sprintf(jumper_symb, "%s_inv", undefsymb->name)` might be better.